### ISSUE TITLE
Unhandled eception from twisted.web._newclient.ResponseNeverReceived 

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#2092]
**Current behavior**
Platform:  (xxxxx.onion)
Version: 2.72.3

twisted.web._newclient.ResponseNeverReceived 
    A L{ResponseFailed} that knows no response bytes at all have been received.
    

Traceback (most recent call last):

  File "/usr/lib/python2.7/dist-packages/globaleaks/jobs/base.py", line 44, in run
    yield self.operation()

ResponseNeverReceived: [<twisted.python.failure.Failure OpenSSL.SSL.Error: [('SSL routines', 'ssl3_read_bytes', 'ssl handshake failure')]>]

**Comment by NSkelsey** [comment]
Issue first seen in v2.71.1 to x

Full list:
- 2.71.1
- 2.71.2
- 2.71.3
- 2.72.2
- 2.72.3

Testcase that demonstrates failure is `twisted/web/client/test/test_newclient.py` named `test_noResponseAtAll`

**Comment by NSkelsey** [comment]
There is a related issue that throws this traceback as well:

```
twisted.web._newclient.ResponseNeverReceived
    A L{ResponseFailed} that knows no response bytes at all have been received.


Traceback (most recent call last):

  File "/usr/lib/python2.7/dist-packages/globaleaks/jobs/base.py", line 45, in run
    yield self.operation()

ResponseNeverReceived: [<twisted.python.failure.Failure twisted.internet.error.ConnectionLost: Connection to the other side was lost in a non-clean fashion: Connection lost.>]
```

I believe handling these exceptions within the `ExternNetLoopingJob` handler will silence both of these failures.

**Comment by NSkelsey** [comment]
This is an occasional failure condition that occurs on outbound connections trying to connect over HTTP. 

Confirmed that it is occurring on outbound jobs.
```
2017-09-25 12:24:59+0000 [-] [E] Job Exit Nodes Refresh died with runtime 4101.1600 [low: 1747.0000, high: 12723.0000]
2017-09-25 12:24:59+0000 [-] Traceback (most recent call last):
2017-09-25 12:24:59+0000 [-]   File &quot;&#47;usr&#47;lib&#47;python2.7&#47;dist-packages&#47;globaleaks&#47;jobs&#47;base.py&quot;, line 57, in run
2017-09-25 12:24:59+0000 [-]     yield self.operation()
2017-09-25 12:24:59+0000 [-] ResponseNeverReceived: [&lt;twisted.python.failure.Failure OpenSSL.SSL.Error: [(&#39;SSL routines&#39;, &#39;ssl3_read_bytes&#39;, &#39;ssl handshake failure&#39;)]&gt;]
```

Specifically during the execution of `ExitNodesRefreshSchedule.operation()` the call to `GLSettings.appstate.tor_exit_set.update(net_agent)` triggers a call into a `twisted.web.client.Agent` via: `request = agent.request('GET', url)`

The agent fetching the request url then uses a `twisted.web._newclient.HTTPClientParser` to handle reading lines of the hosts response to asynchronously interact with a `twisted.web.iweb.IAgent.request` which globaleaks uses to retrieve the body of the HTTP Response.

When `HTTPClientParser.connectionLost` is called before `STATE == DONE` and any data is read on the line, this [failure gets passed](https://github.com/twisted/twisted/blob/twisted-16.0.0/twisted/web/_newclient.py#L548).

I think that it happens much more often over Tor connections, because its very unlikely that the services we are connecting to are accepting TCP connections and then immediately dropping HTTP conns. Instead it is much more likely that the exit nodes are dropping traffic accidentally or maybe intentionally as they try to establish a connection to the service.

**Comment by NSkelsey** [comment]
In fact the second failure that is wrapped in the first provides even more information.

OpenSSL never gets a response:
`[&lt;twisted.python.failure.Failure OpenSSL.SSL.Error: [(&#39;SSL routines&#39;, &#39;ssl3_read_bytes&#39;, &#39;ssl handshake failure&#39;)]&gt;]`

The TCP transport loses its connection:
`twisted.internet.error.ConnectionLost: Connection to the other side was lost in a non-clean fashion: Connection lost`


This is passed through `parser.connectionLost(self, reason)` and passed into `ResponseNeverReceived` via this [line](https://github.com/twisted/twisted/blob/twisted-16.0.0/twisted/web/_newclient.py#L549).

**Comment by NSkelsey** [comment]
Resolved in `ccf84e2`
